### PR TITLE
Datasource-related refactorings of tests (and some utils)

### PR DESCRIPTION
**Pull request by yrodiere**
This is a pre-requisite for #41929, but is only refactorings and addition of more tests, without changing the behavior of Quarkus.

So, I thought it would be best to merge it separately, before #41929. It won't be a waste anyway, as it clarifies some surprising behavior -- which I intend to fix in #41929 -- in the Flyway/Liquibase extensions, as well as reactive clients (see #43517).

### CODE REVIEW

**Comment by yrodiere** [comment]
Also, sorry. Yes it's a lot of tests, but there are so many edge cases, it's a nightmare.

**Review by gsmet — APPROVED** [review]
I spotted a few very minor things. I let you decide if you want to fix them in this PR or in the follow-up. I don't mind either way.

File: extensions/reactive-db2-client/deployment/src/main/java/io/quarkus/reactive/db2/client/deployment/ReactiveDB2ClientProcessor.java [hunk 1]
```diff
@@ -212,40 +210,30 @@ private void createPoolIfDefined(DB2PoolRecorder recorder,
...
         syntheticBeans.produce(db2PoolBeanConfigurator.done());
 
         ExtendedBeanConfigurator mutinyDB2PoolConfigurator = SyntheticBeanBuildItem
                 .configure(io.vertx.mutiny.db2client.DB2Pool.class)
                 .defaultBean()
                 .addType(io.vertx.mutiny.sqlclient.Pool.class)
                 .scope(ApplicationScoped.class)
-                .addInjectionPoint(VERTX_DB2_POOL_TYPE, injectionPointAnnotations(dataSourceName))
+                .qualifiers(qualifiers(dataSourceName))
+                .addInjectionPoint(VERTX_DB2_POOL_TYPE, qualifier(dataSourceName))
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply extensions/reactive-db2-client/deployment/src/main/java/io/quarkus/reactive/db2/client/deployment/ReactiveDB2ClientProcessor.java:228]: Right. IIRC I checked that internally, an empty array gets replaced by `Default`. That was one month ago, but I'm quite confident that was the case.

[the remark this answers is not in the document]
Comment (yrodiere) [reply extensions/reactive-db2-client/deployment/src/main/java/io/quarkus/reactive/db2/client/deployment/ReactiveDB2ClientProcessor.java:228]: Confirmed: a bean declaration with no qualifiers gets the `Default` qualifier automatically.

https://github.com/quarkusio/quarkus/blob/ed8410f7bf5f40468baee5923ea6d93ee60be0c2/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java#L736-L743

File: extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseRuntimeConfig.java [hunk 2]
```diff
@@ -28,7 +29,10 @@ public static LiquibaseRuntimeConfig defaultConfig() {
      * Gets the {@link LiquibaseDataSourceRuntimeConfig} for the given datasource name.
      */
     public LiquibaseDataSourceRuntimeConfig getConfigForDataSourceName(String dataSourceName) {
-        return namedDataSources.getOrDefault(dataSourceName, LiquibaseDataSourceRuntimeConfig.defaultConfig());
+        return DataSourceUtil.isDefault(dataSourceName)
+                ? defaultDataSource
+                : namedDataSources.getOrDefault(dataSourceName,
+                        namedDataSources.getOrDefault(dataSourceName, LiquibaseDataSourceRuntimeConfig.defaultConfig()));
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseRuntimeConfig.java:35]: Good catch; probably an auto-merge gone wrong. Fixed.

File: extensions/reactive-datasource/deployment/src/main/java/io/quarkus/reactive/datasource/deployment/ReactiveDataSourceBuildUtil.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public static AnnotationInstance qualifier(String dataSourceName) {
+        if (dataSourceName == null || DataSourceUtil.isDefault(dataSourceName)) {
+            return AnnotationInstance.builder(Default.class).build();
+        } else {
+            return AnnotationInstance.builder(ReactiveDataSource.class).value(dataSourceName).build();
+        }
+    }
+
+    public static AnnotationInstance[] qualifiers(String dataSourceName) {
+        if (DataSourceUtil.isDefault(dataSourceName)) {
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply extensions/reactive-datasource/deployment/src/main/java/io/quarkus/reactive/datasource/deployment/ReactiveDataSourceBuildUtil.java:24]: Right, I added a commit for that.

**Review by yrodiere — COMMENTED** [review]
Thanks for the review. I addressed all comments, let's merge once CI agrees.

**Comment by yrodiere** [comment]
I forgot about Hibernate Reactive... pushing an update to align it on Hibernate ORM.

**Pull request merged**